### PR TITLE
Fix direct keyboard not working when using a Controller.

### PR DESCRIPTION
**Pull request by MaxLastBreath**
- Pass KeyboardDriver to NpadController.GetHLEKeyboardInput().
- Always fetch a Keyboard with KeyboardDriver.GetGamepad("0") if Direct Keyboard is on.
- Remove unnecessary return null.

### CODE REVIEW

**Review by MelonSpeedruns — APPROVED** [review]
LGTM!

**Review by riperiperi — COMMENTED** [review]
I think you should call `controller.GetHLEKeyboardInput` outside this loop:
`foreach (InputConfig inputConfig in _inputConfig)`
Since it'll repeat the keyboard input for each bound controller (and not work if you don't have any)

What you need to do is something like this:
- Change `public KeyboardInput? GetHLEKeyboardInput(IGamepadDriver KeyboardDriver)` to `public static`. I don't think it uses any non-static members now, but it should be possible without them if it does.
- Call `hleKeyboardInput = NpadController.GetHLEKeyboardInput(_keyboardDriver);` outside the loop, probably after it. Condition should be `(_enableKeyboard && !_blockInputUpdates)`.

**Comment by MaxLastBreath** [comment]
Should be good now, silly me, I didn't even think of the loop.

File: src/Ryujinx.Input/HLE/NpadController.cs [hunk 1]
```diff
@@ -487,36 +487,34 @@ private static Vector3 Truncate(Vector3 value, int decimals)
...
+            {
+                int value = keyboardState.IsPressed(entry.TargetKey) ? 1 : 0;
 
-                return hidKeyboard;
+                hidKeyboard.Modifier |= value << entry.Target;
             }
 
-            return null;
+            return hidKeyboard;
+
```
Comment (riperiperi) [inline src/Ryujinx.Input/HLE/NpadController.cs:517]: Remove extra newline

**Review by riperiperi — APPROVED** [review]
Looks good apart from spacing nit, nice to see this finally fixed. 😌

**Comment by MaxLastBreath** [comment]
> Looks good apart from spacing nit, nice to see this finally fixed. 😌

Removed the extra newline too :D Should be good now, Hopefully. Yeah I'm pretty happy it's fixed, it will be really nice to have it all working in the future.

**Review by MelonSpeedruns — APPROVED** [review]

**Review by riperiperi — APPROVED** [review]

File: src/Ryujinx.Input/HLE/NpadManager.cs [hunk 2]
```diff
@@ -232,10 +232,6 @@ public void Update(float aspectRatio = 1)
 
                         motionState = (controller.GetHLEMotionState(), altMotionState);
```
Comment (gdkchan) [inline src/Ryujinx.Input/HLE/NpadManager.cs:234]:
```suggestion
```

File: src/Ryujinx.Input/HLE/NpadController.cs [hunk 3]
```diff
@@ -487,38 +487,35 @@ private static Vector3 Truncate(Vector3 value, int decimals)
             return value;
         }
 
-        public KeyboardInput? GetHLEKeyboardInput()
+        public static KeyboardInput? GetHLEKeyboardInput(IGamepadDriver KeyboardDriver)
```
Comment (TSRBerry) [inline src/Ryujinx.Input/HLE/NpadController.cs:490]:
```suggestion
        public static KeyboardInput GetHLEKeyboardInput(IGamepadDriver KeyboardDriver)
```
If we never return null there is no need to keep the nullable modifier here.

Comment (MaxLastBreath) [reply src/Ryujinx.Input/HLE/NpadController.cs:490]: That is true, I'm not a keen to how C# works :D, I program in C++ primarily, Thank you for the point.

**Review by gdkchan — APPROVED** [review]
lgtm, thanks.

**Pull request merged**